drivers/memc/memc_rp2350_psram.c

```
@@ -0,0 +1,598 @@
+    while (n--) {
+        /* intentional empty spin */
+        __asm__ volatile("nop");
+    }
+}
+
+/*
+ * Push one byte to the QMI DIRECT_TX FIFO and wait until the transfer
+ * completes.  iwidth selects SPI (0) or Quad (DIRECT_TX_IWIDTH_Q).
+ * Set nopush if you do not want the corresponding RX byte queued.
+ */
+static __ramfunc void psram_tx(uint32_t iwidth, uint8_t data, bool oe, bool nopush) {
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'psram_tx' of convertible types are easily swapped by mistake

(bugprone-easily-swappable-parameters)